### PR TITLE
Context storage

### PR DESCRIPTION
**Pull request by atroyn**

### CODE REVIEW

File: components/context_storage/IContextStorage.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from abc import ABC, abstractmethod
```
Comment (davidbyttow) [inline components/context_storage/IContextStorage.py:1]: This "interface" file seems like it shouldn't have to know anything about the various storage options. Can't we change this such that pinecode/weaviate implement an agnostic interface and contain all of the vector store specifics in their respective files?

File: components/__init__.py
Comment (davidbyttow) [inline components/__init__.py:1]: Delete this directory unless it's actually needed?

File: components/context_storage/IContextStorage.py (hunk 1 above)
Comment (davidbyttow) [reply components/context_storage/IContextStorage.py:1]: Correct. Also, why IContextStorage vs just ContextStorage? I think the latter is more idiomatic as per pep8?

Comment (davidbyttow) [reply components/context_storage/IContextStorage.py:1]: Anyway, happy to help, I think this is a step in the right direction, just need to get it in and we can iterate. @atroyn

Comment (atroyn) [reply components/context_storage/IContextStorage.py:1]: We could in principle take the constructor method out and break this up further, but I'm not sure the modularity is warranted yet.

**Comment by bobvanluijt** [comment]
Awesome PR 💪

You might also consider supporting [Weaviate embedded](https://weaviate.io/developers/weaviate/installation/embedded).

[This](https://github.com/Significant-Gravitas/Auto-GPT#weaviate-setup) is how it's done in AutoGPT, maybe that's helpful.

File: components/context_storage/IContextStorage.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    if storage_type_name == 'pinecone':
+        from .pinecone import Pinecone
+        options = Pinecone.PineconeOptions(storage_name=task_storage_name) if options is None else options
+        return Pinecone(options=options)
+    if storage_type_name == 'weaviate':
+        from .weaviate import Weaviate
+        options = Weaviate.WeaviateOptions(storage_name=task_storage_name) if options is None else options
+        return Weaviate(options=options)
+    else:
+        raise ValueError(f'Invalid storage type: {storage_type_name}
```
Comment (Ryangr0) [inline components/context_storage/IContextStorage.py:46]: ( not ended

**Comment by hsm207** [comment]
@atroyn I work for weaviate and am happy to help review or write (or both!) the weaviate implementation. Let me know how you prefer to collaborate on this PR.

**Pull request closed without merging**